### ISSUE TITLE
Failed to listen on IPv6

### ISSUE THREAD

**Issue by leesei** [ctron/oidc-cli#1]
It may be on me, but I failed to use `oidc-cli` because it tries to bind IPv6 address.

```sh
$ oidc-cli create public keycloak_pub --issuer http://192.168.1.9:8080/realms/my-realm --client-id my-app
10:19:43 [ERROR] Address not available (os error 99)
```

I can use it after changing this line:

```patch
-       let addr = SocketAddr::new(Ipv6Addr::LOCALHOST.into(), port);
+       let addr = SocketAddr::new(Ipv4Addr::LOCALHOST.into(), port);
```

It is sad to see that IPv6 still have such paper cuts nowadays. I've enabled IPv6 at least :stuck_out_tongue_closed_eyes:.
Is there any particular reason for using IPv6?
BTW, I'm using Arch.

**Comment by ctron** [comment]
Interesting, and thanks for reporting.

I would love to have some kind of "just bind to something" :)

I never had any issues with IPv6 in the past, that was the main motivation. I guess having some way to auto-detect this (and fall back) or some `-4` switch would make sense.

Any chance you can come up with a PR for this?

**Comment by leesei** [comment]
I also hoped IPv6 to "just work".

I can submit a PR for binding to IPv4 address.
But adding CLI parameter in Rust program is out of my capability at the moment 😄.

**Comment by leesei** [comment]
Sorry for reviving this issue, I suppose https://github.com/ctron/oidc-cli/blob/main/src/server.rs#L57 is meant to be `(Ipv4Addr::LOCALHOST, port)`?

**Comment by ctron** [comment]
> Sorry for reviving this issue, I suppose https://github.com/ctron/oidc-cli/blob/main/src/server.rs#L57 is meant to be `(Ipv4Addr::LOCALHOST, port)`?

:facepalm: … yes!

**Comment by ctron** [comment]
There's a new release (`0.3.3`) which should really fix the issue :)

**Comment by MartinLoeper** [comment]
Hi @ctron, could you please reconsider adding a `-4` flag to bind the application to an ipv4 socket?

The issue in my case is: I am using devcontainers in vscode and the port forwarding is currently broken (see: https://github.com/microsoft/vscode-remote-release/issues/7029). Thus, I would like to force using the ipv4 address.

**Comment by ctron** [comment]
> Hi @ctron, could you please reconsider adding a `-4` flag to bind the application to an ipv4 socket?

Sounds like a good idea.

**Comment by ctron** [comment]
I released this as `0.4.0`. Maybe you can check and see if that fixes your case too.

**Comment by MartinLoeper** [comment]
Awesome! I just tested it and it works. Thanks so much @ctron!!
